response.go

```diff
@@ -75,12 +75,7 @@ func (r *Response) fixCharset(detectCharset bool, defaultEncoding string) error
 	}
 	contentType := strings.ToLower(r.Headers.Get("Content-Type"))
 
-	if strings.Contains(contentType, "image/") ||
-		strings.Contains(contentType, "video/") ||
-		strings.Contains(contentType, "audio/") ||
-		strings.Contains(contentType, "font/") {
-		// These MIME types should not have textual data.
-
+	if !strings.Contains(contentType, "text/") {
```

Comment:
I agree that the original check is too simplistic, but they're a lot of textual MIME types without `text/` that might have encoding problems as well. There are  `application/xhtml+xml`, `application/xml`, `application/atom+xml`, to name a few.